src/backend/drm/compositor/mod.rs

```diff
@@ -3152,6 +2930,21 @@ where
             true,
         )?;
 
+        if let ScanoutBuffer::Swapchain(slot) = &frame_state
+            .plane_buffer(self.surface.plane())
+            .expect("We have a buffer for the primary plane")
+            .buffer
+        {
+            if slot.format() != element_config.properties.format {
```

Comment:
This needs to be configurable from the outside, I am just not sure how and in what detail. Having this restriction will kill performance on lower end hardware. In my case I would only enable this on general purpose x68 hardware and only when more than a single output is enabled.

Reply:
Hmm, maybe just have this be another bit for `FrameMode`?  We might also have a need for a new flag [here](https://github.com/Smithay/smithay/issues/1610#issuecomment-2546155469). So maybe `FrameMode` should become `FrameFlags` and have something like `ALLOW_PRIMARY_SCANOUT` instead of `PRIMARY_SCANOUT.

Reply:
I have pushed a commit doing this: https://github.com/Smithay/smithay/pull/1576/commits/54379b2e86ed3fbb1b55127bca658706e3128fbe

Reply:
looks fine, thanks!